source/CAS/Client.php

```diff
@@ -942,14 +949,12 @@ public function __construct(
 
         // Make cookie handling available.
         if ($this->isProxy()) {
-            if (!isset($_SESSION['phpCAS'])) {
-                $_SESSION['phpCAS'] = array();
-            }
-            if (!isset($_SESSION['phpCAS']['service_cookies'])) {
-                $_SESSION['phpCAS']['service_cookies'] = array();
+            if (!$this->hasSessionValue('service_cookies')) {
+                $this->setSessionValue('service_cookies', array());
             }
+            // TODO remove explicit call to $_SESSION
             $this->_serviceCookieJar = new CAS_CookieJar(
-                $_SESSION['phpCAS']['service_cookies']
+                $_SESSION[static::PHPCAS_SESSION_PREFIX]['service_cookies']
```

Comment:
I will just keep this as is; the session handler should has been applied here with the reference passing, I guess?